qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -296,8 +298,19 @@ def _postprocess(self, num_samples: int, result: SamplerResult) -> np.ndarray |
             prob = np.zeros((num_samples, *self._output_shape))
 
         for i in range(num_samples):
-            counts = result.quasi_dists[i]
-
+            if isinstance(self.sampler, BaseSamplerV1):
+                counts = result.quasi_dists[i]
+
+            elif isinstance(self.sampler, BaseSamplerV2):
+                bitstring_counts = result[i].data.meas.get_counts()
+                # Normalize the counts to probabilities
+                total_shots = sum(bitstring_counts.values())
+                probabilities = {k: v / total_shots for k, v in bitstring_counts.items()}
+                # Convert to quasi-probabilities
+                counts = QuasiDistribution(probabilities)
+                counts = {k: v for k, v in counts.items() if int(k) < self._output_shape[0]}
+            else:
+                raise QiskitMachineLearningError("Wrong Sampler Type.")
```

Comment:
```suggestion
                raise QiskitMachineLearningError(f"The accepted estimators are BaseSamplerV1 (deprecated) and BaseSamplerV2; got {type(self.sampler)} instead.")
```
Should the `sampler` attribute be private or public?